File: core/src/de/project/ice/ecs/systems/IteratingIceSystem.java
package de.project.ice.ecs.systems;


import com.badlogic.ashley.core.Engine;
import com.badlogic.ashley.core.Family;
import com.badlogic.ashley.systems.IteratingSystem;
import de.project.ice.ecs.IceEngine;

public abstract class IteratingIceSystem extends IteratingSystem
{
    public IteratingIceSystem(Family family)
    {
        super(family);
    }

    public IteratingIceSystem(Family family, int priority)
    {
        super(family, priority);
    }

    @Override
    @Deprecated
    public void addedToEngine(Engine engine)
    {
        addedToEngine((IceEngine) engine);
    }

    @Override
    @Deprecated
    public void removedFromEngine(Engine engine)
    {
        removedFromEngine((IceEngine) engine);
    }

    public void addedToEngine(IceEngine engine)
    {
        super.addedToEngine(engine);
    }

    public void removedFromEngine(IceEngine engine)
    {
        super.removedFromEngine(engine);
    }

    @Override
    public IceEngine getEngine()
    {
        return (IceEngine) super.getEngine();
    }
}

File: core/src/de/project/ice/ecs/systems/IteratingIceSystem.kt
package de.project.ice.ecs.systems


import com.badlogic.ashley.core.Engine
import com.badlogic.ashley.core.Family
import com.badlogic.ashley.systems.IteratingSystem
import de.project.ice.ecs.IceEngine

abstract class IteratingIceSystem : IteratingSystem {
    constructor(family: Family) : super(family) {
    }

    constructor(family: Family, priority: Int) : super(family, priority) {
    }

    @Deprecated("", ReplaceWith("addedToEngine(engine as IceEngine)", "de.project.ice.ecs.IceEngine"))
    override fun addedToEngine(engine: Engine) {
        addedToEngine(engine as IceEngine)
    }

    @Deprecated("", ReplaceWith("removedFromEngine(engine as IceEngine)", "de.project.ice.ecs.IceEngine"))
    override fun removedFromEngine(engine: Engine) {
        removedFromEngine(engine as IceEngine)
    }

    open fun addedToEngine(engine: IceEngine) {
        super.addedToEngine(engine)
    }

    fun removedFromEngine(engine: IceEngine) {
        super.removedFromEngine(engine)
    }

    override fun getEngine(): IceEngine {
        return super.getEngine() as IceEngine
    }
}
